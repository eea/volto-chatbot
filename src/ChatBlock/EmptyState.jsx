import { Button } from 'semantic-ui-react';
import { debounce } from './utils';

function StarterMessage({ msg, onClick }) {
  if (!(msg.name || msg.message)) return null;

  return (
    <Button
      onClick={onClick}
      onKeyDown={onClick}
      role="button"
      tabIndex={-1}
      className="starter-message"
    >
      <span className="starter-message-title">{msg.name || msg.message}</span>
      {/* <div className="starter-message-desc">{msg.description}</div> */}
    </Button>
  );
}

let click_signal = { current: null };

export default function EmptyState(props) {
  const {
    persona,
    onChoice,
    showAssistantPrompts = true,
    enableStarterPrompts,
    starterPrompts = [],
    starterPromptsHeading,
  } = props;

  return (
    <div className="empty-state">
      {enableStarterPrompts &&
        starterPrompts.length > 0 &&
        starterPromptsHeading && (
          <h4 className="starter-message-heading">{starterPromptsHeading}</h4>
        )}

      <div className="starter-messages-container">
        {enableStarterPrompts &&
          starterPrompts.map((msg, idx) => (
            <StarterMessage
              key={msg.name || `starter-${idx}`}
              msg={msg}
              onClick={() =>
                debounce(
                  () =>
                    onChoice(msg.message || `${msg.name}\n${msg.description}`),
                  click_signal,
                )
              }
            />
          ))}

        {showAssistantPrompts &&
          persona?.starter_messages?.map((msg, idx) => (
            <StarterMessage
              key={msg.name || `assistant-${idx}`}
              msg={msg}
              onClick={() =>
                debounce(
                  () =>
                    onChoice(msg.message || `${msg.name}\n${msg.description}`),
                  click_signal,
                )
              }
            />
          ))}
      </div>
    </div>
  );
}
