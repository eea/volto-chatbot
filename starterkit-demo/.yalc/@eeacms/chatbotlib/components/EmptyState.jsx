// import { Button } from 'semantic-ui-react';
import { debounce } from './../utils';

function Button(props) {
  return <button {...props} />;
}

function StarterMessage({ msg, onClick }) {
  return (
    <Button
      onClick={onClick}
      onKeyDown={onClick}
      role="button"
      tabIndex={-1}
      className="starter-message"
    >
      <span className="starter-message-title">{msg.name}</span>
      <div className="starter-message-desc">{msg.description}</div>
    </Button>
  );
}

let click_signal = { current: null };

export default function EmptyState(props) {
  const {
    persona,
    onChoice,
    showAssistantTitle = true,
    showAssistantPrompts = true,
    showAssistantDescription = true,
  } = props;

  return (
    <div className="">
      {showAssistantTitle && <h2>{persona.name}</h2>}
      {showAssistantDescription && <p>{persona.description}</p>}

      <div className="starter-messages-container">
        {showAssistantPrompts &&
          persona.starter_messages?.map((msg) => (
            <StarterMessage
              key={msg.name}
              msg={msg}
              onClick={() =>
                debounce(
                  () =>
                    onChoice(msg.message || `${msg.name}\n${msg.description}`),
                  click_signal,
                )
              }
            />
          ))}
      </div>
    </div>
  );
}
