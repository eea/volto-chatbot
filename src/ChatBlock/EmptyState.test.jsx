import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-intl-redux';
import EmptyState from './EmptyState';

const mockStore = configureStore();

describe('EmptyState', () => {
  it('should render the component', () => {
    const store = mockStore({
      userSession: { token: '1234' },
      intl: {
        locale: 'en',
        messages: {},
      },
    });

    const props = {
      '@type': 'danswerChat',
      assistant: '17',
      chatTitle: 'Online public chat',
      height: '500px',
      onChoice: jest.fn(),
      showAssistantDescription: true,
      showAssistantPrompts: true,
      showAssistantTitle: true,
      persona: {
        name: 'In enim justo rhoncus ut',
        description: 'Nullam dictum felis eu pede',
        starter_messages: [
          {
            description: 'Vestibulum purus quam scelerisque ut ',
            message: 'Nam at tortor in tellus',
            name: 'Curabitur at lacus ac velit',
          },
        ],
      },
    };

    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <EmptyState {...props} />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
