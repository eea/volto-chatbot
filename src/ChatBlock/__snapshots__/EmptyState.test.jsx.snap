// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState should render the component 1`] = `
<div
  className="empty-state"
>
  <div
    className="starter-messages-container"
  >
    <button
      className="ui button starter-message"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={-1}
    >
      <span
        className="starter-message-title"
      >
        Curabitur at lacus ac velit
      </span>
    </button>
  </div>
</div>
`;
